tools/check/check-tidy.sh

```diff
@@ -1,8 +1,15 @@
 #!/usr/bin/env bash
+
+# set is used to set the environment variables.
+#  -e: exit immediately when a command returning a non-zero exit code.
+#  -u: treat unset variables as an error.
+#  -o pipefail: sets the exit code of a pipeline to that of the rightmost command to exit with a non-zero status,
+#       or to zero if all commands of the pipeline exit successfully.
 set -euo pipefail
 
 # go mod tidy do not support symlink
 cd -P .
 
 GO111MODULE=on go mod tidy
-git diff --quiet
+
+git diff --quiet || { echo "The git working tree is not clean." && exit 1; }
```

Comment:
I don't know why we need `git diff --quiet`. Should we remove it?

Reply:
@jackysp It ensures that all the changes we have made are committed.

Reply:
I think it is not necessary. It forces us to commit all changes before `make check`. But `make check` may encounter a format issue, which we have to commit the changes again.

Reply:
@jackysp Updated. PTAL